### PR TITLE
Refactor inference benchmark and add runner script to do sweep

### PR DESCRIPTION
**Pull request by mikaylagawarecki**
- Added `runner.sh` that does a sweep over `batch_size=(1, 32, 64, 128, 256)` and `compile=(True, False)`
- Added GPU utilization as a metric
- Converted frontend from 2 processes (one putting requests into `request_queue` and one reading from `response_queue` and collecting metrics) to a single process with 3 threads (one putting requests into `request_queue` and one reading from `response_queue` and collecting metrics and one polling `nvidia-smi` for gpu utilization)

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #112900
* __->__ #112863
* #112582

### CODE REVIEW

File: benchmarks/inference/results/output.txt [hunk 1]
```diff
@@ -0,0 +1,102 @@
+=============================================================================
```
Comment (albanD) [inline benchmarks/inference/results/output.txt:1]: Do you actually want to checkin this file?

**Review by albanD — APPROVED** [review]
SGTM!

File: benchmarks/inference/results/output.txt (hunk 1 above)
Comment (mikaylagawarecki) [reply benchmarks/inference/results/output.txt:1]: I'll remove it and add the results back when I make the runner output things in tabular form(?)

**Pull request closed without merging**